server/src/routes/comments.js

```
@@ -0,0 +1,123 @@
+/* eslint-disable camelcase */
+import express from 'express'
+import {
+  getComments, getQuestions, getReactions, getThreadComments,
+  getCommentReactions, addReaction, addQuestion, addComment
+} from '../database'
+import { uuid } from './helpers'
+import { catchErrors } from '../handlers'
+
+const router = express.Router()
+
+// @api GET /api/comments
+// Retrieve all comments.
+//
+// returns JSON array of all comments in database
+router.get('/comments', catchErrors(async (req, res) => {
+  await getComments().then((rows) => {
+    res.send(rows)
+  })
+}))
+
+// @api POST /api/comments
+// Adds comment to database.
+//
+// Example body @json {
+//   "text": "minttua",
+//   "user": "salaattipoika",
+//   "blob": "{\"path\": \"/path/to/element\"}"
+// }
+//
+// Returns 'OK' if comment is succesfully added
+router.post('/comments', catchErrors(async (req, res) => {
+  const { text, user, blob } = req.body
+  const id = uuid()
+  const threadId = req.body.threadId || uuid()
+  await addComment({
```

Comment IDs are `CHAR(8)` but passed a full `uuid()` here. It fails or gets silently truncated.

Also need a loop here as insert can fail.

---

No need to truncate `uuid()` is by default 8 long. Loop still needed.